klient/wait/conditions/conditions.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	"context"
+	"log"
+
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apimachinerywait "k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+)
+
+type Condition struct {
+	resources *resources.Resources
+
+	// verbose is used to enable additional logs from the Wait function that is getting executed.
+	// This can help in debugging long-running tests.
+	verbose bool
+}
+
+// New is used to create a new Condition that can be used to perform a series of pre-defined wait checks
+// against a resource in question
+func New(r *resources.Resources) *Condition {
+	return &Condition{resources: r}
+}
+
+// WithVerboseLog is used to enable additional verbose logs from the Wait condition to help debug tests
+func (c *Condition) WithVerboseLog() *Condition {
+	c.verbose = true
```

Comment:
I understand your concern about ability to see messages from long running tests.
I just don't think log control should be done this way using the wait API.
Log should be controlled universally during test runtime and should be handled in a separate PR.

Suggestion:
* Put back the hard coded `log.Println` statement as you had before
* We'll get it merged
* Then I can open an issue to revisit logging universally for the entire project/package

Reply:
If it is all right with you, can we do it the other way? I will cleanup the logs for now and have nothing in it., We can revisit this again with the new Issue and tackle that as a whole thing generally.. 
I will update the PR to cleanup the logging for now.

Reply:
Opened https://github.com/kubernetes-sigs/e2e-framework/issues/77 to track the same.